src/sagas/PermissionsApiSagas.js

```diff
@@ -60,6 +62,44 @@ function* getAclWatcher() :Saga<*> {
   yield takeEvery(GET_ACL, getAclWorker);
 }
 
+/*
+ *
+ * PermissionsApi.getAcls
+ * PermissionsApiActions.getAcls
+ *
+ */
+
+function* getAclsWorker(action :SequenceAction) :Saga<WorkerResponse> {
+
+  if (!isValidAction(action, GET_ACLS)) {
+    return { error: new Error(ERR_INVALID_ACTION) };
+  }
+
+  let workerResponse :WorkerResponse;
+  const { id, value } = action;
+
+  try {
+    yield put(getAcls.request(id, value));
+    const response = yield call(PermissionsApi.getAcls, value);
+    workerResponse = { data: response };
+    yield put(getAcls.success(id, response));
```

Comment:
1. Missing error check?

2. I don't think you have to reassign workerResponse. You could just declare it as a `const` and set `workeResponse.data` and `workerResponse.error` where appropriate

Reply:
1. Error checking does not happen at this level. It happens underneath in `lattice` or above in the apps. `lattice-sagas` is simply a wrapper around `lattice` and it mainly exists to give us the `RequestState` lifecycle. Additionally, `response` here is an `axios` response object, so to check for an error requires extra knowledge / context that we just don't have. We could check for HTTP status codes... but then again, so can the apps themselves.

2. It's because of Flow. Your suggestion triggers errors:

<img width="1025" alt="Screen Shot 2020-09-27 at 9 31 44 AM" src="https://user-images.githubusercontent.com/440299/94370366-8327eb00-00a4-11eb-8a30-cdc19aff108a.png">
<img width="984" alt="Screen Shot 2020-09-27 at 9 31 55 AM" src="https://user-images.githubusercontent.com/440299/94370368-84f1ae80-00a4-11eb-8242-2ee310e662af.png">

I believe this is because of the concepts of "sealed" objects:

https://flow.org/en/docs/types/objects/#toc-sealed-objects

> when objects are sealed, Flow will not allow you to add new properties to them.